### DISCUSSION TITLE
add use cases to (some) example problems

### DISCUSSION THREAD

**Discussion by nikita-petrashen** [pcottle/learnGitBranching#879] [Ideas]
Hey, thanks for your awesome tool! It is tremendously helpful to learn such a complicated thing like Git.

A great addition would be to add example use cases for some of the tasks in the course. Without this one might wonder "Why do I even need to to this?" and the tasks turn into moving colored balls around.

**Comment by pcottle** [comment]
Yep that's a fair point. Usually the lesson intros try to hint at where some of this functionality would be useful, or I try to name the branches in a way that is self-explanatory (like the `bugFix` and `debug` branches).

That being said, I do think there's value in having generalized mobility and fluency within `git`. My thought process is rarely:

- I need to do ABC common task
- ABC common task maps to commands 1,2,3
- Execute 1,2,3

It's more like:
- I'm in situation X
- I need to re-arrange the tree in this specific way to solve my issue
- That re-arranging maps to commands 1,2,3

**Comment by nikita-petrashen** [comment]
Thanks for tour answer!

**Comment by Vokivorob** [comment]
![изображение](https://user-images.githubusercontent.com/99485035/154059483-150c7c03-ca56-476d-b997-4e6a6b2dabce.png)
![изображение](https://user-images.githubusercontent.com/99485035/154059626-4f3c959b-a2ce-4656-ae4b-d784843d7848.png)
Could you please tell me how it should be? I think there's a mistake here.

**Comment by pcottle** [comment]
<img width="901" alt="Screen Shot 2022-02-15 at 7 33 30 AM" src="https://user-images.githubusercontent.com/1135007/154083049-c55f9905-2d5c-49a3-8696-baa6f6cb1da6.png">

You can use `show solution` to get a more guided experience but unfortunately for interactive rebase, we can only tell you the order (not animate the dialog ourselves):
https://learngitbranching.js.org/?NODEMO&command=level%20move2;%20show%20solution

**Reply by Vokivorob** [reply]
You misunderstood me. I have already used <code>show solution</code>, the second screenshot shows the result and it is the same, so something is wrong? Well, or I don't understand you, then I'm sorry.

**Comment by pcottle** [comment]
<img width="1271" alt="Screen Shot 2022-02-15 at 10 53 48 AM" src="https://user-images.githubusercontent.com/1135007/154120383-01a78c78-254e-4f84-a59a-bc4e027016da.png">
oh it should look like this, which is different
